### ISSUE TITLE
Will it run on Raspberry Pi?

### ISSUE THREAD

**Issue by shakedzy** [exo-explore/exo#290]
I couldn't figure out from the README if I can run this on a cluster of Raspberry Pis? Do Linux systems require some sort of Nvidia-based card (which is what I understand from the prerequisites)?

**Comment by AlexCheema** [comment]
Hold off on this until we have PyTorch support merged (#139). You could already try, but others have run into issues and I'm not sure of the root cause.

**Comment by FFAMax** [comment]
> I couldn't figure out from the README if I can run this on a cluster of Raspberry Pis? Do Linux systems require some sort of Nvidia-based card (which is what I understand from the prerequisites)?

By the way do you have a version with PCIE lanes so GPU can be attached?

**Comment by shakedzy** [comment]
> > I couldn't figure out from the README if I can run this on a cluster of Raspberry Pis? Do Linux systems require some sort of Nvidia-based card (which is what I understand from the prerequisites)?
> 
> By the way do you have a version with PCIE lanes so GPU can be attached?

Yes, but I was hoping to use the Hailo AI chip ( https://www.raspberrypi.com/products/ai-kit/ ) rather than an actual GPU

**Comment by dtnewman** [comment]
Indeed, with the current state of [PR 139](https://github.com/exo-explore/exo/pull/139), I have it running on a single raspberry pi. Getting a networking issue with two pis with discovery over UDP, but it works on a single pi (using the Llama 3.2 1b model).
